tests/test_nblist.py

```diff
@@ -376,7 +380,109 @@ def test_nblist_max_interactions(block_size, tiles):
+    else:
+        assert 0, "Invalid host name"
+
+    st = SingleTopology(mol_a, mol_b, core, forcefield)
+    host = setup_optimized_host(st, host_config)
+
+    lambda_grid = np.array([0.0])
+
+    initial_state = setup_initial_states(st, host, DEFAULT_TEMP, lambda_grid, seed=2024, min_cutoff=None)[0]
+
+    return st, host, host_name, initial_state
+
+
+def compute_block_occupancy(
+    nblist, frame: NDArray, box: NDArray, cutoff: float, column_block: int = 32, row_block: int = 32
+) -> float:
+    assert column_block == 32
+    assert column_block == row_block
+    N = frame.shape[0]
+    assert nblist.get_num_row_idxs() == N
+    ixn_list = nblist.get_nblist(frame, box, cutoff)
+    block_occupancies = []
+    max_ixn_per_block = column_block * row_block * ((N + row_block - 1) // row_block)
+    max_ixn_per_block = min(N, max_ixn_per_block)
+
+    for i, ixn_block in enumerate(ixn_list):
+        assert max_ixn_per_block >= len(ixn_block)
+        block_occupancies.append(len(ixn_block) / max_ixn_per_block)
+        # Each interaction group can hold column_block fewer ixns as it is upper triangular
+        max_ixn_per_block -= column_block
+    return float(np.mean([block_occupancies]))
+
+
+@pytest.mark.parametrize("cutoff", [0.9, 1.0, 1.2])
+@pytest.mark.parametrize("precision", [np.float32, np.float64])
+def test_nblist_occupancy_dhfr(cutoff, precision):
+    _, _, coords, box = setup_dhfr()
+
+    if precision == np.float32:
+        nblist = custom_ops.Neighborlist_f32(coords.shape[0])
+    else:
+        nblist = custom_ops.Neighborlist_f64(coords.shape[0])
+    occupancy = compute_block_occupancy(nblist, coords, box, cutoff)
+    assert occupancy > 0.10
+    print(f"DHFR NBList Occupancy with Cutoff {cutoff}: {occupancy * 100.0:.3g}%")
+
+
+@pytest.mark.parametrize("frames", [10])
+@pytest.mark.parametrize("precision", [np.float32])
+def test_nblist_occupancy(hif2a_single_topology_leg, frames, precision):
+    cutoff = 1.2
+    st, host, host_name, initial_state = hif2a_single_topology_leg
+
+    md_params = MDParams(n_eq_steps=0, steps_per_frame=100, n_frames=frames, seed=2024)
+
+    traj = sample(initial_state, md_params, max_buffer_frames=md_params.n_frames)
+
+    if precision == np.float32:
+        nblist = custom_ops.Neighborlist_f32(initial_state.x0.shape[0])
+    else:
+        nblist = custom_ops.Neighborlist_f64(initial_state.x0.shape[0])
+    occupancies = []
+    occupancy = compute_block_occupancy(nblist, initial_state.x0, initial_state.box0, cutoff)
+    occupancies.append(occupancy)
+    for i, (frame, box) in enumerate(zip(traj.frames, traj.boxes)):
+        occupancy = compute_block_occupancy(nblist, frame, box, cutoff)
+        occupancies.append(occupancy)
+    total_steps = md_params.steps_per_frame * md_params.n_frames + md_params.n_eq_steps
+    print(f"Starting occupancy {occupancies[0]}")
+    print(f"Final occupancy after {total_steps:} {occupancies[-1]}")
```

Comment:
Can we assert anything meaningful here?

Reply:
Added two sanity checks, first that the occupancy improves as the barostat runs and that the occupancies are greater than 40%.

https://github.com/proteneer/timemachine/pull/1368/commits/4f6af9dc56144ad0c577461f231b974ce24abde4

Reply:
can we rename this to nblist_tile_density to be more precise instead of occupancy?

Reply:
also, how much does the starting tile density differ from the final frame's tile density?

Reply:
The tile density does not differ by more than about 1%.